suites/create.js

```diff
@@ -308,6 +308,92 @@ export function runDataIntegrityProofFormatTests({
+                'type', 'proofPurpose', 'verificationMethod');
+            }
+          });
+          it.skip('If options has a non-null domain item, it MUST be ' +
+          'equal to proof.domain or an error MUST be raised and SHOULD ' +
+          'convey an error type of PROOF_GENERATION_ERROR.',
+          function() {});
+          it.skip('If options has a non-null challenge item, it MUST ' +
+          'be equal to proof.challenge or an error MUST be raised ' +
+          'and SHOULD convey an error type of PROOF_GENERATION_ERROR.',
+          function() {});
+        });
+        describe('Add Proof Set/Chain', function() {
+          it.skip('Whenever this algorithm encodes strings, ' +
+            'it MUST use UTF-8 encoding.',
+          function() {
+            for(const proof of proofs) {
+              chai.expect(proof.proofValue.isWellFormed()).to.be.true;
+            }
+          });
+          it('If a proof with id equal to previousProof does not ' +
+          'exist in allProofs, an error MUST be raised and SHOULD ' +
+          'convey an error type of PROOF_GENERATION_ERROR.',
+          function() {
+            this.test.link = 'https://www.w3.org/TR/vc-data-integrity/#:~:text=If%20a%20proof%20with%20id%20equal%20to%20previousProofdoes%20not%20exist%20in%20allProofs%2C%20an%20error%20MUST%20be%20raised%20and%20SHOULD%20convey%20an%20error%20type%20of%20PROOF_GENERATION_ERROR.';
+            for(const proof of proofs) {
+              if('previousProof' in proof) {
+                if(typeof proof.previousProof === 'string') {
+                  proof.previousProof = [proof.previousProof];
+                }
+                for(const previousProof in proof.previousProof) {
+                  proofs.some(
+                    otherProof => otherProof.id == previousProof).
+                    should.be('True',
+                      'Expected previousProof value to be the id of another ' +
+                    'included proof.'
+                    );
+                }
+              }
+            }
+          });
+          it('If any element of previousProof array has an id attribute ' +
+          'that does not match the id attribute of any element of allProofs, ' +
+          'an error MUST be raised and SHOULD convey an error type of ' +
+          'PROOF_GENERATION_ERROR.',
+          function() {
+            this.test.link = 'https://www.w3.org/TR/vc-data-integrity/#:~:text=If%20any%20element%20of%20previousProof%20array%20has%20an%20id%20attribute%20that%20does%20not%20match%20the%20id%20attribute%20of%20any%20element%20of%20allProofs%2C%20an%20error%20MUST%20be%20raised%20and%20SHOULD%20convey%20an%20error%20type%20of%20PROOF_GENERATION_ERROR.';
+            const previousProofs = [];
+            for(const proof of proofs) {
+              if('previousProof' in proof) {
+                if(typeof proof.previousProof === 'string') {
+                  proof.previousProof = [proof.previousProof];
```

Comment:
let's not mutate the proof here and instead just add proof specific variable that collects the previous values.

Reply:
I've updated the code, please let me know what you think

Reply:
code looks good here.